articles/building-a-python-package-and-publishing-on-pypi/index.md

```diff
@@ -0,0 +1,169 @@
+
+Before we proceed let’s rearrange for packaging, create a new folder with any name of your choice and put the *init.py* file in the new folder you just created. Then create a file and save it as *setup.py*, this file will hold all the important information the compiler will need to set up our package.
+
+Then add the following code below:
+
+```python
+import pathlib
+from setuptools import setup, find_packages
+
+HERE = pathlib.Path(__file__).parent
+
+VERSION = '0.0.0.1'
+PACKAGE_NAME = 'findlcm'
+AUTHOR = 'james'
+AUTHOR_EMAIL = 'james@sandy.com'
+URL = 'https://github.com/jamessandy/lcmfinder'
+
+LICENSE = 'MIT'
+DESCRIPTION = 'A package that helps find Lowest common multiple of a number'
+LONG_DESCRIPTION = (HERE / "README.md").read_text()
+LONG_DESC_TYPE = "text/markdown"
+
+INSTALL_REQUIRES = [
+      'numpy',
+      'pandas'
+]
+
+setup(name=PACKAGE_NAME,
+      version=VERSION,
+      description=DESCRIPTION,
+      long_description=LONG_DESCRIPTION,
+      long_description_content_type=LONG_DESC_TYPE,
+      author=AUTHOR,
+      license=LICENSE,
+      author_email=AUTHOR_EMAIL,
+      url=URL,
+      install_requires=INSTALL_REQUIRES,
+      packages=find_packages()
+      )
+
+```
+
+The first chunk of code imports all the tools we need to set up our library, the *VERSION* will be the release number for the package, the *PACKAGE NAME* will be a unique name you want to call your package, the *AUTHOR* and *AUTHOR EMAIL* will be your details and finally, the URL can be a site if you have one or maybe your GitHub Repo.
+
+If your package requires other packages to work then you’ll need to indicate them, for example, if you use other packages to write your code you should add them in the *INSTALL_REQUIRES* because your package depends on them (otherwise it will not work) but this package we just created doesn’t require any and I intentionally put wrote this note, in case you’re trying it out with your package idea.
+
+#### Packaging
+Now that we are done writing the code, we will set up all the needed files, and structure the codebase. Another important step will be to sign up for an account on [pypi.org](pypi.org), because that is where we will be hosting our package. After you sign up go to your terminal and pip install Twine with the command below.
+
+```python
+pip install twine
+
+```
+
+The next thing you need to do is navigate to the directory of your project where the setup.py file is and then run this code.
+
+```python
+Python setup.py sdist bdist_wheel
+```
+
+After running the code above you will notice more directories called *dist*, *build*, and *your_package.egg-info*. The *dist* is the most important right now because it holds the installation files that we will deploy to PyPI, open the directory and you’ll see two compressed files that are saved with *.tar* extension and also a wheel file.
```

Comment:
Explain what dist, build directories mean.

Reply:
I've exchanged directory with folder, the I'veexplained dist,